redis/src/main/scala/zio/redis/api/Keys.scala

```diff
@@ -211,10 +209,13 @@ trait Keys {
    */
   final def scan(
     cursor: Long,
-    pattern: Option[Regex] = None,
+    pattern: Option[String] = None,
     count: Option[Long] = None,
     `type`: Option[String] = None
-  ): ZIO[RedisExecutor, RedisError, (Long, Chunk[String])] = Scan.run((cursor, pattern, count, `type`))
+  ): ZIO[RedisExecutor, RedisError, (Long, Chunk[String])] =
+    Task.effect(`type`.map(RedisType.from)).refineOrDie { case e: RedisError => e } >>= (redisType =>
```

Comment:
Let's accept RedisType instead and remove this parsing step completely.

Reply:
Change to RedisType